draft-ietf-ppm-dap.md

```diff
@@ -812,47 +834,49 @@ at least twice the cache lifetime in order to avoid rejecting reports.
 
 ### Upload Request
 
-Clients upload reports by using an HTTP POST to `[leader]/upload`, where
-`[leader]` is the first entry in the task's aggregator endpoints. The payload is
-structured as follows:
+Clients upload reports by using an HTTP PUT to
+`[leader]/tasks/{task-id}/reports/{report-id}`, where `[leader]` is the first
+entry in the task's aggregator endpoints.
+
+`task-id` is the ID of the task to which the report is being uploaded.
+`report-id` is a `ReportId` used by the Aggregators to ensure the report appears
+in at most one batch (see {{input-share-validation}}). The Client MUST generate
+this using a cryptographically secure random number generator.
+
+[Note/TODO: it's important that the _Client_ chooses the report ID. We use the
+report ID as the nonce, and VDAF-04 will include some guidance explaining that
+if the report ID is chosen with knowledge of the VDAF verify key, privacy could
+be violated. For that reason, we can't allow either aggregator to choose report
+IDs.]
+
+The payload is structured as follows:
 
 ~~~
 struct {
-  ReportID report_id;
```

Comment:
Hrmm, I'm a bit wary. of moving the report ID out of the `Report` struct. Can you explain your thinking here?

In any case, we need to make sure that the task ID, report ID, timestamp, and any other metadata that we might have in the future are all included in the report metadata.

Reply:
The report ID now occurs in the request path `[leader]/tasks/{task-id}/reports/{report-id}`, so it does not need to occur in the request body. `struct ReportMetadata` does still contain `ReportID` so that it can be used to construct `InputShareAad` and `ReportShare`.

I suppose it's not essential that the report ID be in the request path: thus far, it's only the task ID that's needed by servers to figure out how to parse the rest of the message. So we could put report ID back into the body, and then we can use `ReportMetadata` in `Report` again, which is a little nicer.

Reply:
The downside of this change is that it's inconsistent with the URIs for aggregation jobs and collections, but I don't really think that matters.

Reply:
My vote would be to move report ID back into body so that the metadata appears in both report and report share. If you decide to to keep the report ID where it is, then my only request is to make sure to spell out how the metadata is constructed where we specify encryption of the input shares.

Reply:
Oh looks like you already updated this! Thanks.